src/edu/psu/sweng500/team8/play/ActionManager.java

```diff
@@ -36,12 +36,13 @@ public void doUndo(CellGrid currentCellGridFromBoard) {
 		 * revert last action on the sudokuActionQueue, and put that action into
 		 * sudokuActionQueueForUndo
 		 */
-		SudokuAction lastAction = sudokuActionQueue.pop();
 		
-		if (null != lastAction) {
-			CellGrid previousCellGrid = lastAction.getCellGrid();
-			lastAction.setCellGrid(new CellGrid(currentCellGridFromBoard));
-			sudokuActionQueueForUndo.add(lastAction);
+		if (null != sudokuActionQueue.peek()) {

```

Comment:
This might be a bit nitpicky...Typically "queue" suggests a first-in-first-out (FIFO) queue. I would call it a Stack to avoid confusion.